aptos-move/block-executor/src/executor.rs

```diff
@@ -695,8 +772,11 @@ where
             )
         };
 
-        if matches!(ret, Err(Error::ModulePathReadWrite)) {
-            debug!("[Execution]: Module read & written, sequential fallback");
+        if matches!(
+            ret,
+            Err(Error::ModulePathReadWrite) | Err(Error::ResourceGroupError)
```

Comment:
Do we have to change executor.rs code to fall back to sequential execution on ResourceGrroupErrors?

Reply:
it's a safety measure, if there is a ResourceGroupError that we caught and it was due to concurrency, this would allow us to gracefully recover.